docs/source/format/ADBC/Go.rst

```diff
@@ -0,0 +1,31 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+=========================
+ADBC Go API Specification
+=========================
+
+In Go, ADBC consists of a set of interface definitions in the package
+``github.com/apache/arrow-adbc/go/adbc``.
+
+Broadly, the interfaces are organized similarly to the C API
+specification, and bindings to the C API can be created easily.
+
+See apache/arrow-adbc commit f044edf5256abfb4c091b0ad2acc73afea2c93c0_
+for the definitions.
```

Comment:
Two questions: 

1. Can we rename the `LICENSE.txt` in the root to just `LICENSE`? Doing so should allow Go to propagate the license to the nested module correctly (otherwise we'd need to symlink or otherwise put a `LICENSE.txt` or otherwise into that dir.
2. After doing that, we should be able to point to https://pkg.go.dev/github.com/apache/arrow-adbc/go/adbc to contain the definitions and documentation.

Reply:
This is for the format - I was thinking we would vote on and tag the actual code release separately, if that's OK, and then we can rename the license.

Reply:
That's fair, I was just thinking it makes sense to point to the interfaces via docs rather than by commit.